devtools/create_project/xcode.cpp

```diff
@@ -25,6 +25,7 @@
 
 #include <fstream>
 #include <algorithm>
+#include <sys/stat.h>
```

Comment:
> Does this even work? It looks like it's for static library linking, which takes a full path. I doubt we can simply supply every library and the non-existent ones only cause warnings.

It would work for the paths which are used for `-I` and `-L`. However, the linker will emit warnings for non-existing directories passed to `-L`, which is annoying. I was trying to avoid these.

Library linking for ScummVM OS X uses “normal” `-l` via `OTHER_LDFLAGS` (`scummvmOSX_LdFlags`) instead of putting them into Xcode’s special Linked Frameworks and Libraries section with `DEF_LOCALLIB_STATIC`. The `DEF_LOCALLIB_STATIC` is used only for iOS, and iOS is busted anyway, so, uh. …

> Might work, still doesn't help when we pre-generate project files AFAICT.

Can you tell me more about what the scenario is where these files are pre-generated to be given to other users? To me, at the moment, this feels like asking for a way to give people pre-generated Makefiles without running autoconf on the build machine, which I don’t think is possible either (unless the two machines are the same, of course)?

> Yeah, you will need to implement it yourself in devtools/ code. ScummVM main code cannot be linked to devtools binaries.

So if the runtime directory detection were maintained, I would need to implement a Win32 branch? Are there other platforms that need to be considered?

Practically, and perhaps naïvely, I would think just wrapping in `#ifndef _WIN32` would be good enough; I have no idea why anyone would reasonably try to generate Xcode projects on Windows (and the only practical reason why I could think of someone doing it on a platform other than Mac OS would be for unit testing the Xcode generator). Is this unreasonable?